src/lbaf/IO/configValidator.py

```diff
@@ -0,0 +1,123 @@
+        self.__logger = logger
+
+    @staticmethod
+    def is_valid(valid_schema: Schema, schema_to_validate: dict) -> bool:
+        """ Returns True is schema_to_validate is valid with self.valid_schema else False. """
+        is_valid = valid_schema.is_valid(schema_to_validate)
+        return is_valid
+
+    @staticmethod
+    def validate(valid_schema: Schema, schema_to_validate: dict):
+        """ Returns validated schema. """
+        return valid_schema.validate(schema_to_validate)
+
+    def main(self):
+        """ Main routine for the config validation. """
+        # Validate Skeleton
+        if self.is_valid(valid_schema=self.__skeleton, schema_to_validate=self.__config_to_validate):
+            self.__logger.info("Skeleton schema is VALID!")
+        else:
+            self.__logger.error("Skeleton schema is INVALID!")
+            self.validate(valid_schema=self.__skeleton, schema_to_validate=self.__config_to_validate)
+
+        # Validate from_data/from_samplers
+        if (from_data := self.__config_to_validate.get("from_data")) is not None:
+            self.__logger.info("Reading from data was chosen.")
+            if self.is_valid(valid_schema=self.__from_data, schema_to_validate=from_data):
+                self.__logger.info("from_data schema is VALID!")
+            else:
+                self.__logger.error("from_data schema is INVALID!")
+                self.validate(valid_schema=self.__from_data, schema_to_validate=from_data)
+        elif (from_samplers := self.__config_to_validate.get("from_samplers")) is not None:
+            self.__logger.info("Simulate from samplers was chosen.")
+            if self.is_valid(valid_schema=self.__from_samplers, schema_to_validate=from_samplers):
+                self.__logger.info("from_samplers schema is VALID!")
+            else:
+                self.__logger.error("from_samplers schema is INVALID!")
+                self.validate(valid_schema=self.__from_samplers, schema_to_validate=from_samplers)
+
+        # Validate Algorithm
+        if (algorithm := self.__config_to_validate.get("algorithm").get("name")) is not None:
+            self.__logger.info(f"Checking algorithm: {algorithm}")
+            if self.is_valid(valid_schema=self.__algorithm.get(algorithm),
+                             schema_to_validate=self.__config_to_validate.get("algorithm")):
+                self.__logger.info(f"Algorithm: {algorithm} schema is VALID!")
+            else:
+                self.__logger.error("Algorithm: {algorithm} schema is INVALID!")
+                self.validate(valid_schema=self.__algorithm.get(algorithm),
+                              schema_to_validate=self.__config_to_validate.get("algorithm"))
+
+
+if __name__ == "__main__":
+    with open("../Applications/conf.yaml", "rt") as config_file:
+        yaml_str = config_file.read()
+        configuration = yaml.safe_load(yaml_str)
+
+    ConfigValidator(config_to_validate=configuration, logger=logger()).main()
```

Comment:
Please use entire names : `configuration_validator` or `ConfigurationValidator`, but not `ConfigValidator`.